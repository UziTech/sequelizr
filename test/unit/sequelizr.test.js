const util = require("util");
const exec = util.promisify(require("child_process").exec);

function sequelizr(argsString) {
	return exec(`node ../../bin/sequelizr.js ${argsString}`, {cwd: __dirname});
}

const {version} = require("../../package.json");

describe("sequelizr", () => {
	test("should show help -h", async () => {
		const {stdout} = await sequelizr("-h");
		expect(stdout).toMatchSnapshot();
	});

	test("should show help --help", async () => {
		const {stdout} = await sequelizr("--help");
		expect(stdout).toMatchSnapshot();
	});

	test("should show version -v", async () => {
		const {stdout} = await sequelizr("-v");
		expect(stdout.trim()).toBe(version);
	});

	test("should show version --version", async () => {
		const {stdout} = await sequelizr("--version");
		expect(stdout.trim()).toBe(version);
	});
});
