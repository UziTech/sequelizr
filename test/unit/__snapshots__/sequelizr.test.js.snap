// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sequelizr should show help --help 1`] = `
"sequelizr <cmd> [opts]

Commands:
  sequelizr check [opts]     Check if models match the database tables.
  sequelizr download [opts]  Save tables to models.
  sequelizr upload [opts]    Create tables from models.

Global Options:
  -h, --help     Show help  [boolean]
  -v, --version  Show version number  [boolean]

Examples:
  sequelizr <cmd> --help  Show args for a command.
"
`;

exports[`sequelizr should show help -h 1`] = `
"sequelizr <cmd> [opts]

Commands:
  sequelizr check [opts]     Check if models match the database tables.
  sequelizr download [opts]  Save tables to models.
  sequelizr upload [opts]    Create tables from models.

Global Options:
  -h, --help     Show help  [boolean]
  -v, --version  Show version number  [boolean]

Examples:
  sequelizr <cmd> --help  Show args for a command.
"
`;
